### PR TITLE
Fix creating partition is not atomic

### PR DESCRIPTION
**Pull request by bigsheeper**
issues: https://github.com/milvus-io/milvus/issues/23436

/kind bug

### CODE REVIEW

**Review by longjiquan — CHANGES_REQUESTED** [review]
Please also add recovery logic for creating partition.

File: internal/rootcoord/garbage_collector.go [hunk 1]
```diff
@@ -147,6 +148,20 @@ func (c *bgGarbageCollector) ReDropPartition(pChannels []string, partition *mode
 	_ = redo.Execute(context.Background())
 }
 
+func (c *bgGarbageCollector) RemoveCreatingPartition(partition *model.Partition, ts Timestamp) {
+	redoTask := newBaseRedoTask(c.s.stepExecutor)
+
+	redoTask.AddAsyncStep(&removePartitionMetaStep{
```
Comment (longjiquan) [inline internal/rootcoord/garbage_collector.go:154]: `removePartitionMetaStep` should be executed after `releasePartition`.

**Review by longjiquan — CHANGES_REQUESTED** [review]

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**